test/.gitignore

```diff
@@ -0,0 +1,12 @@
+# Ignore everything
+*
+
+# ...even if they are in subdirectories
+!*/
```

Comment:
I think you are trying to ignore anything in subdir, so shouldn't it be as below?
*/